### PR TITLE
Improve slot based OATH with touch

### PR DESCRIPTION
**Pull request by dagheyman**
Add workaround for bug in NEO, where the device will blink when it's not supposed to. Also improve logging a bit.

### CODE REVIEW

File: ykman/driver_otp.py [hunk 1]
```diff
@@ -123,12 +126,16 @@ def _read_pid(self):
...
-        if ykpers.yk_get_serial(self._dev, 0, 0, byref(serial)):
-            return serial.value
-        else:
-            return None  # Serial not visible
+        for i in range(1, attempts):
+            if ykpers.yk_get_serial(self._dev, 0, 0, byref(serial)):
+                return serial.value
+            else:
+                logger.debug(
+                    'Failed to read serial from device. Attempt {}.'.format(i))
```
Comment (emlun) [inline ykman/driver_otp.py]: The logger methods have formatting built in - see review summary.

File: ykman/driver_otp.py [hunk 2]
```diff
@@ -286,6 +293,10 @@ def calculate(
         # Give the YubiKey 10 tries to do the calculation.
         for idx in range(10):
             try:
+                logger.debug(
+                    'Sending a challenge to the device. Slot {}. '
+                    'Attempt {}. Wait for touch is {}.'.format(
+                        slot, idx, wait_for_touch))
```
Comment (emlun) [inline ykman/driver_otp.py]: Ditto

File: ykman/driver_otp.py [hunk 3]
```diff
@@ -294,8 +305,17 @@ def calculate(
                     # Error 11 when wait_for_touch is true is an unexpected
                     # state, let's try again.
                     continue
+                elif wait_for_touch is False:
+                    logger.debug('Got {} as expected.'.format(e))
+                    # NEOs and very old YK4s might still be blinking,
+                    # lets try to read the serial to cancel it.
+                    self._read_serial()
+                    raise
```
Comment (emlun) [inline ykman/driver_otp.py]: Ditto

File: ykman/driver_otp.py [hunk 4]
```diff
@@ -294,8 +305,17 @@ def calculate(
...
                 else:
+                    logger.debug('YkpersError: {}'.format(e))
```
Comment (emlun) [inline ykman/driver_otp.py]: Ditto

File: ykman/driver_otp.py (hunk 1 above)
Comment (dagheyman) [reply ykman/driver_otp.py]: Will update the logging, thx.

**Comment by dagheyman** [comment]
Commit updated

**Pull request merged**